File: Python/String/4_rotation_of_string.py
"""
Idea:
using 2 instances of a string and concatenating them,
checking if other string is in the concatenated string.

"""


def areRotations(string1, string2):
    sumString = string1 + string1
    if string2 in sumString:
        return True
    return False


# Driver code
if __name__ == "__main__":
    string1 = "AABCD"
    string2 = "BACDA"

    # Function call
    if areRotations(string1, string2):
        print("Strings are rotations of each other")
    else:
        print("Strings are not rotations of each other")

# This code is contributed by Bhavya Jain

File: C++/String/4_rotation_of_string.cpp
/*
Idea:
using 2 instances of a string and concatenating them,
checking if other string is in the concatenated string.

*/
#include <bits/stdc++.h>
using namespace std;

bool areRotations(string str1, string str2)
{
    string sumString = str1 + str1;
    if (str1.size() != str2.size())
        return 0;
    return sumString.find(str2) != string::npos;
}

int main()
{
    string str1 = "AACD", str2 = "ACDA";

    // Function call
    if (areRotations(str1, str2))
        printf("Strings are rotations of each other");
    else
        printf("Strings are not rotations of each other");
    return 0;
}
